src/main/java/org/bananarama/crud/DeleteOperation.java

```diff
@@ -53,4 +53,11 @@
      * @return
      */
     public DeleteOperation<T> from(Stream<T> data,QueryOptions options);
+    
+    /**
+     * Delete all records from the persisted layer.
+     * @param data The collection of objects to be removed
+     * @return
+     */
+    public DeleteOperation<T> all();
```

Comment:
Ok with breaking backwards compatibility (this is an API extension and as such should break existing Adapter implementation but not client code). There must be a similar method with QueryOptions arg for consistency with similar methods for reading data. @totomz let's avoid default implementations that throw UnsupportedOperationException.